src/Storages/System/StorageSystemProjections.cpp

```diff
@@ -108,13 +112,15 @@ class ProjectionsSource : public ISource
                     continue;
 
                 const auto table = tables_it->table();
-                if (!table)
+                if (!table || !table->isMergeTree())
                     continue;
                 StorageMetadataPtr metadata_snapshot = table->getInMemoryMetadataPtr();
                 if (!metadata_snapshot)
                     continue;
                 const auto & projections = metadata_snapshot->getProjections();
 
+                auto parts = dynamic_cast<const MergeTreeData &>(*table).getDataPartsVectorForInternalUsage({MergeTreeData::DataPartState::Active}, nullptr);
```

Comment:
Why not to use `typeid` cast? You've checked that the table is a MergeTree in the code above.